source/MaterialXRenderGlsl/TextureBaker.cpp

```diff
@@ -0,0 +1,114 @@
+//
+// TM & (c) 2019 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXRenderGlsl/TextureBaker.h>
+
+#include <MaterialXRenderGlsl/GLTextureHandler.h>
+#include <MaterialXRenderGlsl/GlslValidator.h>
+
+#include <MaterialXRender/LightHandler.h>
+#include <MaterialXRender/StbImageLoader.h>
+
+#include <MaterialXFormat/XmlIo.h>
+
+namespace MaterialX
+{
+
+void TextureBaker::bakeShaderInputs(ShaderRefPtr shaderRef, const FileSearchPath& searchPath, GenContext& context, const FilePath& outputFolder)
+{
+    if (!shaderRef)
+    {
+        return;
+    }
+
+    _searchPath = searchPath;
+    for (BindInputPtr bindInput : shaderRef->getBindInputs())
+    {
+        OutputPtr output = bindInput->getConnectedOutput();
+        if (output)
+        {
+            bakeGraphOutput(output, context, outputFolder);
+        }
+    }
+}
+
+void TextureBaker::bakeGraphOutput(OutputPtr output, GenContext& context, const FilePath& outputFolder)
+{
+    if (!output)
+    {
+        return;
+    }
+
+    _rasterizer = GlslValidator::create(_frameBufferRes);
```

Comment:
This can still be initialized outside and passed in as an argument as it should not vary for each bake
As mentioned before, I'm just a bit worried about the overhead of recreating the window+gl context every time..

Reply:
Good point, and I've modified the logic so that the rasterizer is now initialized in the TextureBaker constructor.